src/js/containers/recipient/topFive/TopFiveContainer.jsx

```diff
@@ -0,0 +1,153 @@
+import { isCancel } from 'axios';
+
+import {
+    getTrailingTwelveMonths,
+    convertFYToDateRange
+} from 'helpers/fiscalYearHelper';
+import * as SearchHelper from 'helpers/searchHelper';
+import BaseStateCategoryResult from 'models/v2/state/BaseStateCategoryResult';
+
+import TopFive from 'components/recipient/topFive/TopFive';
+
+const propTypes = {
+    total: PropTypes.number,
+    category: PropTypes.string,
+    fy: PropTypes.string,
+    recipientHash: PropTypes.string
+};
+
+export class TopFiveContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+            error: false,
+            results: []
+        };
+
+        this.request = null;
+    }
+
+    componentDidMount() {
+        this.loadCategory();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.recipientHash !== this.props.recipientHash) {
+            this.loadCategory();
+        }
+        if (prevProps.fy !== this.props.fy) {
+            this.loadCategory();
+        }
+    }
+
+    dataParams() {
+        let timePeriod = null;
+        if (this.props.fy === 'latest') {
+            const trailing = getTrailingTwelveMonths();
+            timePeriod = {
+                start_date: trailing[0],
+                end_date: trailing[1]
+            };
+        }
+        else if (this.props.fy !== 'all' && this.props.fy) {
+            const range = convertFYToDateRange(parseInt(this.props.fy, 10));
+            timePeriod = {
+                start_date: range[0],
+                end_date: range[1]
+            };
+        }
+
+        const filters = {
+            id: this.props.recipientHash
+        };
+
+        if (timePeriod) {
+            filters.time_period = [timePeriod];
+        }
+
+        return {
+            filters,
+            category: this.props.category,
+            limit: 5,
+            page: 1
+        };
+    }
+
+    loadCategory() {
+        if (this.request) {
+            this.request.cancel();
+        }
+
+        this.setState({
+            loading: true,
+            error: false
+        });
+
+        this.request = SearchHelper.performSpendingByCategorySearch(this.dataParams());
+        this.request.promise
+            .then((res) => {
+                this.parseResults(res.data.results, res.data.category);
+            })
+            .catch((err) => {
+                if (!isCancel(err)) {
+                    console.log(err);
+                    this.setState({
+                        loading: false,
+                        error: true
+                    });
+                }
+            });
+    }
+
+    parseResults(data, type) {
+        const parsed = data.map((item, index) => {
+            const result = Object.create(BaseStateCategoryResult);
+            result.populate(item, index + 1);
+
+            // use a special naming template for DUNS
+            if (type === 'recipient_duns') {
```

Comment:
There is no DUNS table for the recipient profile, so this block can be removed. However, you will want to look at the mockup and set the naming template for anything that doesn't follow the default `code - name` (found in BaseStateCategoryResult).

Reply:
See https://github.com/fedspendingtransparency/usaspending-website/blob/dev/src/js/containers/state/topFive/TopFiveContainer.jsx#L151 for agency/ subagency name template